stumpy/stump_tiles.py

```diff
@@ -0,0 +1,643 @@
+
+    σ_Q_inverse : ndarray
+        Inverse standard deviation of the query sequence, `Q`, relative to the current
+        sliding window
+
+    cov_a : ndarray
+        The first covariance term relating T_A[i + k + m - 1] and M_T_m_1[i + k]
+
+    cov_b : ndarray
+        The second covariance term relating T_B[i + m - 1] and μ_Q_m_1[i]
+
+    cov_c : ndarray
+        The third covariance term relating T_A[i + k - 1] and M_T_m_1[i + k]
+
+    cov_d : ndarray
+        The fourth covariance term relating T_B[i - 1] and μ_Q_m_1[i]
+
+    μ_Q_m_1 : ndarray
+        Mean of the query sequence, `Q`, relative to the current sliding window and
+        using a window size of `m-1`
+
+    T_A_subseq_isfinite : ndarray
+        A boolean array that indicates whether a subsequence in `T_A` contains a
+        `np.nan`/`np.inf` value (False)
+
+    T_B_subseq_isfinite : ndarray
+        A boolean array that indicates whether a subsequence in `T_B` contains a
+        `np.nan`/`np.inf` value (False)
+
+    T_A_subseq_isconstant : ndarray
+        A boolean array that indicates whether a subsequence in `T_A` is constant (True)
+
+    T_B_subseq_isconstant : ndarray
+        A boolean array that indicates whether a subsequence in `T_B` is constant (True)
+
+    tiles : ndarray
+        An array of two column arrays representing each tile
+
+    tiles_ranges : ndarray
+        A two-dimensional array representing tile indices for each thread
+
+    thread_idx : int
+        The thread index
+
+    ρ : ndarray
+        The Pearson correlations
+
+    I : ndarray
+        The matrix profile indices
+
+    ignore_trivial : bool
+        Set to `True` if this is a self-join. Otherwise, for AB-join, set this to
+        `False`. Default is `True`.
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    `DOI: 10.1007/s10115-017-1138-x \
+    <https://www.cs.ucr.edu/~eamonn/ten_quadrillion.pdf>`__
+
+    See Section 4.5
+
+    The above reference outlines a general approach for traversing the distance
+    matrix in a diagonal fashion rather than in a row-wise fashion.
+
+    `DOI: 10.1145/3357223.3362721 \
+    <https://www.cs.ucr.edu/~eamonn/public/GPU_Matrix_profile_VLDB_30DraftOnly.pdf>`__
+
+    See Section 3.1 and Section 3.3
+
+    The above reference outlines the use of the Pearson correlation via Welford's
+    centered sum-of-products along each diagonal of the distance matrix in place of the
+    sliding window dot product found in the original STOMP method.
+    """
+    m_inverse = 1.0 / m
+    constant = (m - 1) * m_inverse * m_inverse  # (m - 1)/(m * m)
+    chunk_size = config.STUMPY_DIAGONALS_CHUNK_SIZE
+
+    for tile_idx in tiles_ranges[thread_idx]:
+        # break if tile index is negative
+        if tile_idx < 0:
+            break
+
+        y_offset, x_offset = tiles[tile_idx][0]
+        tile_height, tile_width = tiles[tile_idx][1]
+        diags = np.arange(tiles[tile_idx][2][0], tiles[tile_idx][2][1])
```

Comment:
This creates a new array every iteration and, independently, by every thread. This will be very expensive and should be avoided. You'll want to move the creation of these arrays outside of the for-loop and then reuse them by overwriting its elements.

Reply:
Done!